karapace/metrics.py

```diff
@@ -0,0 +1,103 @@
+"""
+karapace - metrics
+Supports collection of system metrics
+list of supported metrics:
+connections-active - The number of active HTTP(S) connections to server.
+                     Data collected inside aiohttp request handler.
+
+Copyright (c) 2023 Aiven Ltd
+See LICENSE for details
+"""
+from __future__ import annotations
+
+from karapace.base_stats import StatsClient
+from karapace.config import Config
+from karapace.prometheus import PrometheusClient
+from karapace.statsd import StatsdClient
+
+import threading
+
+
+class MetricsException(Exception):
+    pass
+
+
+class Singleton(type):
+    _instance: Singleton | None = None
+
+    def __call__(cls, *args: str, **kwargs: int) -> Singleton:
+        if cls._instance is None:
+            instance = super().__call__(*args, **kwargs)
+            cls._instance = instance
+        return cls._instance
+
+
+class Metrics(metaclass=Singleton):
+    stats_client: StatsClient
```

Comment:
`stats_client: StatsClient | None = None`, I've run this locally with `metrics_extended` set to `False` and its failing because the `stats_client` its not going to be set

Reply:
Fixed and unit test for this case is added.